tools/chapel-py/setup.py

```diff
@@ -47,6 +47,41 @@
 
 host_bin_subdir = str(chpl_variables.get("CHPL_HOST_BIN_SUBDIR"))
 chpl_lib_path = os.path.join(chpl_home, "lib", "compiler", host_bin_subdir)
+# For installations using --prefix, the lib final lib path are going to differ
+# figure it out now and write it to the rpath
+chpl_install_lib_path = None
+if os.path.exists(os.path.join(chpl_home, "configured-prefix")):
+    with open(os.path.join(chpl_home, "CMakeLists.txt"), "r") as f:
+        # read CMakeLists.txt to get the CHPL_MAJOR_VERSION and
+        # CHPL_MINOR_VERSION and then construct the path from that
+        chpl_major_version = None
+        chpl_minor_version = None
+        for line in f:
+            if "set(CHPL_MAJOR_VERSION" in line:
+                chpl_major_version = line.split()[1].strip(")")
+            if "set(CHPL_MINOR_VERSION" in line:
+                chpl_minor_version = line.split()[1].strip(")")
+            if (
+                chpl_major_version is not None
+                and chpl_minor_version is not None
+            ):
+                break
+    assert chpl_major_version is not None and chpl_minor_version is not None
+    chpl_version_string = "{}.{}".format(
+        chpl_major_version,
+        chpl_minor_version,
+    )
+    chpl_prefix = None
+    with open(os.path.join(chpl_home, "configured-prefix"), "r") as f:
+        chpl_prefix = f.read().strip()
+    assert chpl_prefix is not None
+    chpl_install_lib_path = os.path.join(
+        chpl_prefix,
+        "lib",
+        "chapel",
+        chpl_version_string,
+        "compiler",
+    )
```

Comment:
In general, I don't think this logic belongs here. I think it should be in something like `util/chplenv/chpl_home_utils.py`, and can be accessed as `python util/chplenv/chpl_home_utils.py --configured-install-prefix`

That said, the logic looks fine to me